extendr-api/src/robj.rs

```diff
@@ -1868,4 +1916,29 @@ mod tests {
         assert_eq!(p, Robj::from(2));
         Ok(())
     }
+
+    #[test]
+    fn str_iter_test() -> Result<(), AnyError> {
+        start_r();
+        let obj = Robj::from(vec!["a", "b", "c"]);
+        assert_eq!(obj.str_iter().unwrap().collect::<Vec<_>>(), vec!["a", "b", "c"]);
+
```

Comment:
It would be good to also add a test that explicitly tests conversion from a factor. Note that the dataframe example uses factors for R 3.x but not for 4.x, so going forward it doesn't test the factor case.